bugzilla_dashboard/people.py

```diff
@@ -38,129 +34,105 @@ def get_access_token(iam_credentials):
+    org = {}
+    emails = {}
+
     # Retrieve access token from IAM
     iam_token = get_access_token(iam_credentials)
 
-    # Retrieve full payloads with that token
-    data = get_all_info(iam_token)
-
-    all_cns = {}
-    all_dns = {}
-
-    new_data = {}
-    for person in data["users"]:
-        person = person["profile"]
-        if not person["access_information"]["hris"]["values"]:
-            continue
-        mail = person["access_information"]["hris"]["values"]["primary_work_email"]
-        dn = person["identities"]["mozilla_ldap_id"]["value"]
-        manager_mail = person["access_information"]["hris"]["values"][
-            "managers_primary_work_email"
-        ]
-        if not manager_mail:
-            manager_mail = mail
-
-        _mail = person["identities"]["mozilla_ldap_primary_email"]["value"]
-        assert mail == _mail
-
-        ismanager = person["staff_information"]["manager"]["value"]
-        isdirector = person["staff_information"]["director"]["value"]
-        cn = "{} {}".format(person["first_name"]["value"], person["last_name"]["value"])
-        bugzillaEmail = ""
-        if "bugzilla_mozilla_org_primary_email" in person["identities"]:
-            bugzillaEmail = person["identities"]["bugzilla_mozilla_org_primary_email"][
-                "value"
-            ]
-        if not bugzillaEmail and "HACK#BMOMAIL" in person["usernames"]["values"]:
-            bugzillaEmail = person["usernames"]["values"]["HACK#BMOMAIL"]
-
-        if bugzillaEmail is None:
-            bugzillaEmail = ""
-
-        del person["usernames"]["values"]["LDAP-posix_id"]
-        del person["usernames"]["values"]["LDAP-posix_uid"]
-        im = list(person["usernames"]["values"].values())
-
-        title = person["staff_information"]["title"]["value"]
-        all_cns[mail] = cn
-        all_dns[mail] = dn
-
-        new = {
-            "mail": mail,
-            "manager": {"cn": "", "dn": manager_mail},
-            "ismanager": "TRUE" if ismanager else "FALSE",
-            "isdirector": "TRUE" if isdirector else "FALSE",
-            "cn": cn,
-            "dn": dn,
-            "bugzillaEmail": bugzillaEmail,
-            "title": title,
+    # Browse full org with that token
+    for user in get_all_users(iam_token):
+
+        # Build a simplified profile containing:
+        # * id
+        # * name
+        # * email
+        # * bugzilla Email
+        # * manager Email
+        # * picture
+        profile = user["profile"]
+        org[user["id"]] = {
+            "id": user["id"],
+            "name": f"{profile['first_name']} {profile['last_name']}",
+            "email": profile["primary_email"],
+            "bugzillaEmail": profile["identities"][
+                "bugzilla_mozilla_org_primary_email"
+            ],
+            "manager": profile["access_information"]["hris"][
+                "managers_primary_work_email"
+            ],
+            "picture": profile["picture"],
```

Comment:
Do we need that ?

Reply:
We can display the avatar on the frontend, makes for a nicer interface

Reply:
ok